SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the test
+	ASpatialComponentTest* CompTest = CastChecked<ASpatialComponentTest>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTest::StaticClass(), FTransform(FVector(-250, -250, 0))));
+
+	// Add the helpers, as we need things placed in the level
+	CompTest->LevelActor = CastChecked<ASpatialComponentTestActor>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTestActor::StaticClass(), FTransform(FVector(-250, -250, 0))));
+	CompTest->LevelReplicatedActor = CastChecked<ASpatialComponentTestReplicatedActor>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTestReplicatedActor::StaticClass(), FTransform(FVector(-250, -250, 0))));
+
+	// Quirk of the test. We need the player spawns on the same portion of the map as the test, so they are LBed together
+	AActor** PlayerStart = CurrentLevel->Actors.FindByPredicate([](AActor* Actor) {
+		return Actor->GetClass() == APlayerStart::StaticClass();
+	});
+	(*PlayerStart)->SetActorLocation(FVector(-500, -250, 100));
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest1x2WorkerSettings::StaticClass());
+	WorldSettings->DefaultGameMode = ASpatialAuthorityTestGameMode::StaticClass();
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir()
+														   + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast/SpatialComponentMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// Custom made for the SpatialTestHandover, placed in the slow category
+void UGenerateTestMapsCommandlet::CreateSpatialHandoverMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the tests
+	// This test is here, because I cannot run it a 100 times in a row
+	GEditor->AddActor(CurrentLevel, ASpatialTestHandover::StaticClass(), FTransform::Identity);
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest2x2WorkerSettings::StaticClass());
+	WorldSettings->bEnableDebugInterface =
+		false; // The test directly accesses the LayeredStrategy and expects it to be the top one, setting to false for now
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(
+		FPaths::ProjectContentDir() + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Slow_Spatial_Only/SpatialHandoverMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+int32 UGenerateTestMapsCommandlet::Main(const FString& CmdLineParams)
+{
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Generate test maps commandlet started. Pray."));
+
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Creating the SpatialNetworkingMap."));
+	CreateSpatialNetworkingMap();
```

Comment:
Interesting. We could do that, however right now the majority of tests do not require more than their own actor, so I feel kind of bad imposing this structure on every test if most will only specify their own actor at 0,0,0 in there.

Also, there's a deeper philosophical point about the structure here, which I think I will have to think over and probably post in Slack.

Reply:
True - I do think we can try making something by default such as
```
virtual FMapDefinition ProvideTestMapDefinition()
{
    return FMapDefinition{{ GetClass(), FVector::Zero }};
}
```
to avoid this kind of setup. But yeah, it's a big departure from what you seem to have had in mind, and you absolutely have a point in making the maps in the way you've specified.